platform-hub-api/spec/integration/tokens_deletion_integration_spec.rb

```diff
@@ -0,0 +1,292 @@
+  let!(:cluster_to_project_allocation) do
+    create :allocation, allocatable: cluster, allocation_receivable: project
+  end
+  let!(:user_group) { create :kubernetes_group, :not_privileged, :for_user }
+  let!(:robot_group) { create :kubernetes_group, :not_privileged, :for_robot }
+  let!(:user_group_to_project_allocation) do
+    create :allocation, allocatable: user_group, allocation_receivable: project
+  end
+  let!(:robot_group_to_service_allocation) do
+    create :allocation, allocatable: robot_group, allocation_receivable: service
+  end
+  let!(:user_token_groups) { [ user_group.name ] }
+  let!(:user_token) do
+    create :user_kubernetes_token, tokenable: identity, project: project, cluster: cluster, groups: user_token_groups
+  end
+  let!(:robot_token_groups) { [ robot_group.name ] }
+  let!(:robot_token) do
+    create :robot_kubernetes_token, tokenable: service, cluster: cluster, groups: robot_token_groups
+  end
+
+  # Other tokens associated with the project and service, but different everything else
+  let!(:other_cluster) { create :kubernetes_cluster }
+  let!(:other_cluster_to_project_allocation) do
+    create :allocation, allocatable: other_cluster, allocation_receivable: project
+  end
+  let!(:other_user_group) { create :kubernetes_group, :not_privileged, :for_user }
+  let!(:other_robot_group) { create :kubernetes_group, :not_privileged, :for_robot }
+  let!(:other_user_group_to_project_allocation) do
+    create :allocation, allocatable: other_user_group, allocation_receivable: project
+  end
+  let!(:other_robot_group_to_service_allocation) do
+    create :allocation, allocatable: other_robot_group, allocation_receivable: service
+  end
+  let!(:other_user_token_groups) { [ other_user_group.name ] }
+  let!(:other_user_token) do
+    create :user_kubernetes_token, project: project, cluster: other_cluster, groups: other_user_token_groups
+  end
+  let!(:other_robot_token_groups) { [ other_robot_group.name ] }
+  let!(:other_robot_token) do
+    create :robot_kubernetes_token, tokenable: service, cluster: other_cluster, groups: other_robot_token_groups
+  end
+
+  # Other tokens that belong to different projects/services
+  let!(:other_other_user_token) { create :user_kubernetes_token }
+  let!(:other_other_robot_token) { create :robot_kubernetes_token }
+
+  def exists model, id
+    expect(model.exists?(id: id)).to be true
+  end
+
+  def deleted model, id
+    expect(model.exists?(id: id)).to be false
+  end
+
+  describe 'on project deletion' do
+    before { project.destroy }
+
+    it 'should delete the associated token(s) only' do
+      deleted KubernetesToken, user_token.id
+      deleted KubernetesToken, robot_token.id
+
+      deleted KubernetesToken, other_user_token.id
+      deleted KubernetesToken, other_robot_token.id
+
+      exists KubernetesToken, other_other_user_token.id
+      exists KubernetesToken, other_other_robot_token.id
+    end
+  end
+
+  describe 'on project membership deletion' do
```

Comment:
`on user's project membership deletion` to make it clearer?

Reply:
Good idea, will update.